### PR TITLE
fix(TimelineNav): remove unused props and styles

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

- update related snapshots

### Test Plan:

- [x] verify that existing tests pass
- [x] update any snapshots

### CODE REVIEW

**Review by ahuth — APPROVED** [review]

File: src/components/TimelineNav/TimelineNav.module.css [hunk 1]
```diff
@@ -280,7 +280,6 @@
   height: fit-content;
   padding: var(--eds-size-half) 0;
   /* Provides white background to fake a visual gap in the vertical line between nav items. */
-  background: var(--eds-theme-color-background-neutral-default);
```
Comment (booc0mtaco) [inline src/components/TimelineNav/TimelineNav.module.css:283]: the visual diffs were generated b/c there was an embedded bg around the icons that was inconsistently transparent. Removing the bg fixes this bug
Screen Shot 2023-01-10 at 17 59 56

Comment (booc0mtaco) [reply src/components/TimelineNav/TimelineNav.module.css:283]: Before:
Screen Shot 2023-01-10 at 18 20 01

**Review by jinlee93 — APPROVED** [review]
Thanks for the cleanup!

**Comment by booc0mtaco** [comment]
I noticed some other oddities in the styling and behavior as compared to what's in figma, so expanded on this PR a little bit to simplify the CSS. Do more with less :) (should get a second look tho... right now making final adjustments to align properly and get the right diff in chromatic)

File: src/components/TimelineNav/TimelineNav.tsx [hunk 2]
```diff
@@ -389,16 +371,14 @@ export const TimelineNav = ({
...
-            {
-              [styles['timeline-nav__list--ordered']]: variant === 'ordered',
-            },
           )}
           role="tablist"
         >
           {timelineNavItems().map((tab, i) => {
             const isActive = activeIndexState === i;
-            const itemVariant = variant && tab.props.variant;
+            const itemVariant = tab.props.variant;
```
Comment (booc0mtaco) [inline src/components/TimelineNav/TimelineNav.tsx:381]: @jinlee93 the chromatic test got to this line, which now will always have a value. previously, `variant` being undefined would set `itemVariant` to undefined i think

Comment (jinlee93) [reply src/components/TimelineNav/TimelineNav.tsx:381]: That does make sense

**Pull request merged**